cmd/explaintest/r/explain_generate_column_substitute.result

```diff
@@ -0,0 +1,532 @@
+└─IndexScan_5	10.00	cop[tikv]	table:t, index:e, range:[3,3], keep order:false, stats:pseudo
+select b+a from t where b+a=3;
+b+a
+3
+desc select e from t where b+a=3;
+id	count	task	operator info
+IndexReader_6	10.00	root	index:IndexScan_5
+└─IndexScan_5	10.00	cop[tikv]	table:t, index:e, range:[3,3], keep order:false, stats:pseudo
+select e from t where b+a=3;
+e
+3
+desc select a+1 from t where a+1 in (1, 2, 3);
+id	count	task	operator info
+IndexReader_6	30.00	root	index:IndexScan_5
+└─IndexScan_5	30.00	cop[tikv]	table:t, index:c, range:[1,1], [2,2], [3,3], keep order:false, stats:pseudo
+select a+1 from t where a+1 in (1, 2, 3);
+a+1
+1
+1
+2
+3
+desc select * from t where a+1 in (1, 2, 3);
+id	count	task	operator info
+IndexLookUp_10	30.00	root	
+├─IndexScan_8	30.00	cop[tikv]	table:t, index:c, range:[1,1], [2,2], [3,3], keep order:false, stats:pseudo
+└─TableScan_9	30.00	cop[tikv]	table:t, keep order:false, stats:pseudo
+select * from t where a+1 in (1, 2, 3);
+a	b	c	e
+1	2	2	3
+2	2.1	3	4.1
+0	0	1	0
+0	0	1	0
+desc select a+1 from t where a+1 between 1 and 4;
+id	count	task	operator info
+IndexReader_6	250.00	root	index:IndexScan_5
+└─IndexScan_5	250.00	cop[tikv]	table:t, index:c, range:[1,4], keep order:false, stats:pseudo
+select a+1 from t where a+1 between 1 and 4;
+a+1
+1
+1
+2
+3
+desc select * from t where a+1 between 1 and 4;
+id	count	task	operator info
+IndexLookUp_10	250.00	root	
+├─IndexScan_8	250.00	cop[tikv]	table:t, index:c, range:[1,4], keep order:false, stats:pseudo
+└─TableScan_9	250.00	cop[tikv]	table:t, keep order:false, stats:pseudo
+select * from t where a+1 between 1 and 4;
+a	b	c	e
+1	2	2	3
+2	2.1	3	4.1
+0	0	1	0
+0	0	1	0
+desc select * from t group by a+1;
+id	count	task	operator info
+HashAgg_5	8000.00	root	group by:test.t.c, funcs:firstrow(test.t.a)->test.t.a, funcs:firstrow(test.t.b)->test.t.b, funcs:firstrow(test.t.c)->test.t.c, funcs:firstrow(test.t.e)->test.t.e
+└─TableReader_11	10000.00	root	data:TableScan_10
+  └─TableScan_10	10000.00	cop[tikv]	table:t, range:[-inf,+inf], keep order:false, stats:pseudo
+select * from t group by a+1;
+a	b	c	e
+1	2	2	3
+2	2.1	3	4.1
+5	3	6	8
+0	0	1	0
+-1	-2	0	-3
+desc select a+1 from t group by a+1;
+id	count	task	operator info
+HashAgg_5	8000.00	root	group by:test.t.c, funcs:firstrow(test.t.c)->Column#3
+└─IndexReader_12	10000.00	root	index:IndexScan_11
+  └─IndexScan_11	10000.00	cop[tikv]	table:t, index:c, range:[NULL,+inf], keep order:false, stats:pseudo
+select a+1 from t group by a+1;
+a+1
+0
+1
+2
+3
+6
+desc select b, avg(a+1) from t group by a+1;
+id	count	task	operator info
+Projection_4	8000.00	root	test.t.b, Column#6
+└─HashAgg_5	8000.00	root	group by:Column#11, funcs:avg(Column#9)->Column#6, funcs:firstrow(Column#10)->test.t.b
+  └─Projection_18	10000.00	root	cast(test.t.c)->Column#9, test.t.b, test.t.c
+    └─TableReader_11	10000.00	root	data:TableScan_10
+      └─TableScan_10	10000.00	cop[tikv]	table:t, range:[-inf,+inf], keep order:false, stats:pseudo
```

Comment:
This plan is worse than before. 
```
mysql> create table t(a int, b real);
Query OK, 0 rows affected (0.02 sec)
mysql> set sql_mode="";
Query OK, 0 rows affected (0.01 sec)
mysql> desc select b, avg(a+1) from t group by a+1;
+-------------------------+----------+------+------------------------------------------------------------------------------+
| id                      | count    | task | operator info                                                                |
+-------------------------+----------+------+------------------------------------------------------------------------------+
| Projection_4            | 8000.00  | root | test.t.b, 2_col_0                                                            |
| └─HashAgg_9             | 8000.00  | root | group by:col_3, funcs:avg(col_0, col_1), firstrow(col_2)                     |
|   └─TableReader_10      | 8000.00  | root | data:HashAgg_5                                                               |
|     └─HashAgg_5         | 8000.00  | cop  | group by:plus(test.t.a, 1), funcs:avg(plus(test.t.a, 1)), firstrow(test.t.b) |
|       └─TableScan_8     | 10000.00 | cop  | table:t, range:[-inf,+inf], keep order:false, stats:pseudo                   |
+-------------------------+----------+------+------------------------------------------------------------------------------+
5 rows in set (0.00 sec)
```

Reply:
This is because we can't support agg push down，and I will comment the related code until we support agg push down.